acceptance/tests/facts/windows_os.rb

```diff
@@ -0,0 +1,91 @@
+test_name "Test facter os reports correct os.windows.system32" do
+    require 'timeout'
+    require 'socket'
+
+    confine :to, :platform => /windows/
+
+    @initial_boot = nil
+
+    def is_host_up(host, port = 22, timeout = nil)
```

Comment:
this logic seems way to complicated. Beaker allows you to restart the node and then check if it is back up
`host.reboot` and `host.up?` which should work on any platform
I made a minimal test that should show how those can be used.

```
def host_is_up?(agent:, max_retries:)
  retries = 0
  begin
    agent.up?
    puts("host is up")
    true
  rescue Errno::ETIMEDOUT
    puts("retry #{retries}")
    if retries <= max_retries
      retries += 1
      sleep 2 ** retries
      retry
    else
      false
    end
  end
end


test_name 'test restart' do
  step 'restart' do
    agent.reboot
  end

  step 'check if host is up' do
    fail_test 'not up' unless host_is_up?(agent: agent, max_retries: 3)
  end

  step 'execute test' do
    on(agent, "cmd.exe /c echo %time%") do |result|
      refute_empty(result.stdout.chomp)
    end
  end
end
```